### ISSUE TITLE
dev is broken because sourced-ce no longer reads docker-compose.override

### ISSUE THREAD

**Issue by dpordomingo** [src-d/sourced-ui#318]
caused by https://github.com/src-d/sourced-ce/pull/262

Since global `docker-compose.override.yml` is no longer linked in source{d} workdirs
(see [sourced-ce#262::factory.go::L119](https://github.com/src-d/sourced-ce/pull/262/files#diff-d9b57abf31b07aa5b4030f940edba729L119)),
it can no longer be read from `~/.sourced/compose-files/__active__/docker-compose.override.yml`
(see [sourced-ui::Makefile::L13](https://github.com/src-d/sourced-ui/blob/master/Makefile#L13)).

**Comment by smacker** [comment]
originally when proposing to remove `override.yml` support from srcd-ce I supposed we will change ui makefile as well by putting file into `~/.sourced/workdirs/__active__/` instead of `~/.sourced/compose-files/__active__/`.

**Comment by dpordomingo** [comment]
Putting the `override` inside of `~/.sourced/workdirs/__active__` may be not enough because it's a link to a directory which does not exist before source{d} has been initialized, so it would require extra steps in certain order to run the dev mode.

I think it could be an initial step, but I think we should consider a better DX in the future.

**Comment by se7entyse7en** [comment]
So at this moment, the flow would required to first run `init`, so that the directory is there, and then call `make set-override` that will put inside `~/.sourced/workdirs/__active__`. And this should work, right?

> I think it could be an initial step, but I think we should consider a better DX in the future.

Yup, it's not very convenient. But I'd first fix the thing and update the doc, and then create another issue to improve it. Unless you guys already have an idea on how to fix this.

**Comment by dpordomingo** [comment]
I think it should be something like this:
```bash
$ make build-dev
$ sourced init local .
$ make dev-prepare
$ sourced restart
```

But then we should also add a
```yml
version: '3.4'
services:
  sourced-ui-celery:
    restart: "no"
```
into `docker-compose.override.yml` to avoid a restart loop in `sourced-ui-celery` after restarting  source{d} in dev mode.

**Comment by dpordomingo** [comment]
So yes, we can do it, close this issue, and open a new one to enhance the process which is suboptimal.

**Comment by dpordomingo** [comment]
Also, we should document that it should be done that way **every time** the user wants to change to a different workdir, because in the past, it was needed to run `make dev-prepare` only once (the first time)

**Comment by se7entyse7en** [comment]
> So yes, we can do it, close this issue, and open a new one to enhance the process which is suboptimal.

But now it doesn't work, right? I won't close this issue then. Or I misunderstood something?

**Comment by dpordomingo** [comment]
Yes, it's currently broken; I meant that I agree with your comment:
>_But I'd first fix the thing and update the doc, and then create another issue to improve it._

We can close after fixing and documenting, and then opening a new one for improving later.

**Comment by se7entyse7en** [comment]
Ah ok ok! yup! Thanks!
